thoth/storages/ceph.py

```diff
@@ -97,6 +99,15 @@ def retrieve_blob(self, object_key: str) -> bytes:
                 raise NotFoundError("Failed to retrieve object, object {!r} does not exist".format(object_key)) from exc
             raise
 
+    def retrieve_document_last_modification_date(self, object_key: str) -> datetime:
+        """Retrieve the last modification date of a document from S3."""
+        try:
+            return self._s3.Object(self.bucket, f"{self.prefix}{object_key}").get()["LastModified"]
+        except botocore.exceptions.ClientError as exc:
+            if exc.response["Error"]["Code"] in ("404", "NoSuchKey"):
+                raise NotFoundError("Failed to retrieve object, object {!r} does not exist".format(object_key)) from exc
+            raise
```

Comment:
Do we really need to catch exceptions at that level ?

Since we're re-raising anyway (-> so the caller has to handle it itself), I would drop that.